src/Client.php

```diff
@@ -542,21 +559,67 @@ protected function send(array $data, array $tags = [])
      */
     protected function sendMessages(array $messages)
     {
+        if (is_null($this->socket)) {
+            $this->socket = $this->connect();
+        }
+        $this->message = implode("\n", $messages);
+        $this->write();
+
+        return $this;
+    }
+
+    /**
+     * Attempt to write the current message to the socket
+     *
+     * @return bool
+     */
+    private function write()
+    {
+        if (!is_null($this->socket)) {
+            if (@fwrite($this->socket, $this->message) === false) {
+                // attempt to re-send on socket resource failure
+                $this->socket = $this->connect();
+                if (!is_null($this->socket)) {
+                    return (@fwrite($this->socket, $this->message) !== false);
+                }
+            } else {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Creates a persistent connection to the udp host:port
+     *
+     * @return resource
+     * @throws ConnectionException If there is a connection problem with the host
+     */
+    protected function connect()
+    {
         $socket = @fsockopen('udp://' . $this->host, $this->port, $errno, $errstr, $this->timeout);
-        if (!$socket) {
+        if ($socket === false) {
+            $socket = null;
```

Comment:
That alternatives are not great either. But I guess safer:

1. If connection fails set to false and never try and re-connect [simple, but loosing stats as soon as there is a network blip]
1. If connection fails have some logic to connect only on a exponentially increasing time curve [more complex]